corehq/warehouse/transforms/sql/location_dim.sql

```diff
@@ -0,0 +1,130 @@
+INSERT INTO {{ location_dim }} (
+    domain,
+
+    location_id,
+    sql_location_id,
+
+    level,
+    location_level_0,
+    location_level_1,
+    location_level_2,
+    location_level_3,
+    location_level_4,
+    location_level_5,
+    location_level_6,
+    location_level_7,
+
+    name,
+    site_code,
+    external_id,
+    supply_point_id,
+    is_archived,
+    latitude,
+    longitude,
+
+    location_type_name,
+    location_type_code,
+    location_type_id,
+
+    location_last_modified,
+    location_created_on,
+
+    dim_last_modified,
+    dim_created_on,
+    -- TODO: Figure out how to handle deletes when we actually hard deleted the model
+    deleted
+)
+WITH RECURSIVE location_hierarchy(
```

Comment:
interesting, i've never seen this syntax. when you had asked about a recursive query i thought you mean just joining the table to itself.

also does this work when we pull in a child location to the staging table but not its parent if the parent wasn't modified in the window?